File: src/main/java/com/vjiki/music/service/StoryServiceImpl.java
package com.vjiki.music.service;

import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

import org.springframework.stereotype.Service;

import com.vjiki.music.dto.StoryResponse;
import com.vjiki.music.mapper.StoryMapper;
import com.vjiki.music.repository.StoryRepository;

@Service
public class StoryServiceImpl implements StoryService {

    private final StoryRepository storyRepository;

    public StoryServiceImpl(StoryRepository storyRepository) {
        this.storyRepository = storyRepository;
    }

    @Override
    public List<StoryResponse> getStoriesByUserId(UUID userId) {
        OffsetDateTime now = OffsetDateTime.now();
        return storyRepository.findActiveStoriesByUserId(userId, now).stream()
                .map(StoryMapper::toResponse)
                .toList();
    }
}

File: bin/main/com/vjiki/music/service/StoryServiceImpl.kt
package com.vjiki.music.service

import com.vjiki.music.dto.StoryResponse
import com.vjiki.music.mapper.StoryMapper.toResponse
import com.vjiki.music.repository.StoryRepository
import org.springframework.stereotype.Service
import java.time.OffsetDateTime
import java.util.UUID

@Service
class StoryServiceImpl(
    private val storyRepository: StoryRepository
) : StoryService {

    override fun getStoriesByUserId(userId: UUID): List<StoryResponse> {
        val now = OffsetDateTime.now()
        return storyRepository.findActiveStoriesByUserId(userId, now)
            .map { it.toResponse() }
    }
}

